### ISSUE TITLE
ncatted not honoring pre-existing history global attribute

### ISSUE THREAD

**Issue by pnorton-usgs** [nco/nco#252]
I have a set of netcdf files where there is a global attribute named 'History'. I did some processing (variable attribute editing, rechunking, etc) on these files followed by concatenation to a single file using various NCO commands (ncatted, ncks, ncrcat). The final file correctly contains all the processing steps inserted into the pre-existing History global attribute. I'm now trying out pynco and discovered that the history entries include too much path information because of the way it calls the various NCO commands. I decided to process the files without writing history and then update the history at the very end of the workflow with sanitized/simplified history entries. While figuring out what that command might look like I tried a line like the following at the command line:

```
ncatted -h -a "history","global",a,c,"ncrcat file_1.nc file_2.nc -o outfile.nc\n" outfile.nc
```

The source files contain the pre-existing global attribute named 'History' but the resulting file ended up with a new global attribute named 'history' along with the original global attribute named 'History'. The manual indicates that while 'history' is the correct form for this attribute, NCO will use the first instance of that word (e.g. History, HISTORY, etc) when writing the history information (p. 145 in nco_5.0.5-alpha01 manual). I'm not sure what to make of this; is this a bug in ncatted (other tools seem to honor the case insensitivity) or is this by design? I wasn't able to find any other reports of this behavior.

I am using NCO version 4.9.7 for my experiments. I appreciate any insight/help you can provide.

**Comment by czender** [comment]
This is by design. A feature not a bug, I hope. `ncatted` follows the NCO manual for generating its own history. And when you specifically tell it to append to 'history' not 'History' it does what you tell it to do. So try changing your command above to
`ncatted -h -a "History","global",a,c,"ncrcat file_1.nc file_2.nc -o outfile.nc\n" outfile.nc
`

**Comment by czender** [comment]
A prepend mode seems like a good idea. I'll add it to the wishlist for 5.0.6.

**Comment by czender** [comment]
FYI prepend mode was easy to add, and you can now try it in `ncatted` in newly released NCO 5.0.5. From the release notes:

> C. ncatted has a new edit mode: Prepend. Analogous to append mode,
> prepend mode will prepend the user-specified value to the attribute
> and if the attribute does not exist it will be created:
> 
> ncatted -a char_att,att_var,p,c,"this is prepended.\n" in.nc out.nc
> 
> http://nco.sf.net/nco.html#att_prepend

Feedback welcome!
